library/extractor/src/test/java/com/google/android/exoplayer2/extractor/avi/DataHelper.java

```diff
@@ -0,0 +1,199 @@
+  }
+
+  public static StreamFormatBox getAacStreamFormat() throws IOException {
+    final Context context = ApplicationProvider.getApplicationContext();
+    final byte[] buffer = TestUtil.getByteArray(context,"extractordumps/avi/aac_stream_format.dump");
+    final ByteBuffer byteBuffer = ByteBuffer.wrap(buffer);
+    byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+    return new StreamFormatBox(StreamFormatBox.STRF, buffer.length, byteBuffer);
+  }
+
+  public static StreamFormatBox getVideoStreamFormat() {
+    final ByteBuffer byteBuffer = AviExtractor.allocate(40);
+    final VideoFormat videoFormat = new VideoFormat(byteBuffer);
+    videoFormat.setWidth(720);
+    videoFormat.setHeight(480);
+    videoFormat.setCompression(VideoFormat.XVID);
+    return new StreamFormatBox(StreamFormatBox.STRF, byteBuffer.capacity(), byteBuffer);
+  }
+
+  public static ListBox getVideoStreamList() {
+    final StreamHeaderBox streamHeaderBox = getVidsStreamHeader();
+    final StreamFormatBox streamFormatBox = getVideoStreamFormat();
+    final ArrayList<Box> list = new ArrayList<>(2);
+    list.add(streamHeaderBox);
+    list.add(streamFormatBox);
+    return new ListBox(streamHeaderBox.getSize() + streamFormatBox.getSize(),
+        ListBox.TYPE_STRL, list);
+  }
+
+  public static ListBox getAacStreamList() throws IOException {
+    final StreamHeaderBox streamHeaderBox = getAudioStreamHeader();
+    final StreamFormatBox streamFormatBox = getAacStreamFormat();
+    final ArrayList<Box> list = new ArrayList<>(2);
+    list.add(streamHeaderBox);
+    list.add(streamFormatBox);
+    return new ListBox((int)(streamHeaderBox.getSize() + streamFormatBox.getSize()),
+        ListBox.TYPE_STRL, list);
+  }
+
+  public static StreamNameBox getStreamNameBox(final String name) {
+    byte[] bytes = name.getBytes();
+    bytes = Arrays.copyOf(bytes, bytes.length + 1);
+    return new StreamNameBox(StreamNameBox.STRN, bytes.length, ByteBuffer.wrap(bytes));
+  }
+
+  public static ByteBuffer appendNal(final ByteBuffer byteBuffer, byte nalType) {
+    byteBuffer.put((byte)0);
+    byteBuffer.put((byte)0);
+    byteBuffer.put((byte) 1);
+    byteBuffer.put(nalType);
+    return byteBuffer;
+  }
+
+  public static ChunkHandler getVideoChunkHandler(int sec) {
+    final FakeTrackOutput fakeTrackOutput = new FakeTrackOutput(false);
+    return new ChunkHandler(0, ChunkHandler.TYPE_VIDEO, fakeTrackOutput,
+        new ChunkClock(sec * 1_000_000L, sec * FPS));
+  }
+
+  public static ChunkHandler getAudioChunkHandler(int sec) {
+    final FakeTrackOutput fakeTrackOutput = new FakeTrackOutput(false);
+    return new ChunkHandler(AUDIO_ID, ChunkHandler.TYPE_AUDIO, fakeTrackOutput,
+        new ChunkClock(sec * 1_000_000L, sec * FPS * AUDIO_PER_VIDEO));
+  }
+
+  public static AviSeekMap getAviSeekMap() {
+    final int[] keyFrameOffsetsDiv2= {4, 1024};
+    final UnboundedIntArray videoArray = new UnboundedIntArray();
```

Comment:
I spent some time on this.  I didn't see anything in Guava that was a full drop-in replacement.  The array does grow dynamically and it feels clunky to move the logic from UnboundIntArray into AviSeekMap.  As a compromise, we could replace UnboundIntArray with ArrayList&lt;Integer&gt; and make a call to Ints.toArray(), that would be more overhead on the front end, but keep long term structures efficient.  Thoughts?

Reply:
I'm not super clear on why a regular array would not work. Otherwise, you can also look into an IntBuffer.

Reply:
Let me give you all the detail.  The seek array is a subset of key frames.  For AVI, the key frame flag is stored in the index (idx1).  I don't know the number of key frames until after I have parsed the index.  As such, I need a dynamic array (like Java List).  There are two sets of arrays.  A single array that manages the seek position and a multidimensional array that manages the timestamp of the streams (usually 1 video and 1 audio).  Both int[] and IntBuffer are fixed length.  UnboundIntArray acts similar to List in that it manages the unbounded nature much as List does.

Part of the confusion is that I'm using parallel arrays, which I know is not a great design pattern.  It is very memory efficient.  A more OO implementation might be a List<> with an Object that has the attributes position: long and chunkIndex:int[n] where n is the number of streams in the file.

@AquilesCanta If you want to have a quick Duo call we can do it in the morning.  I'm generally available after 8:30am MST, except Weds

Reply:
FYI:  MP4/ISO and AVI are amazing similar, so I spent some time looking at the MP4 stuff to see how they got around this.  Their ChunkIndex is akin to my AviSeekMap (parallel arrays and all).  It appears they know the size of their index so that's how they get around this problem, so it's a different situation.  No help there.

Reply:
@AquilesCanta Alright, I think I have something that meets both of our goals.  I moved UnboundIntArray into AviExtractor.  This allowed me to simplify the interface and remove the guard code.  My intention was that it could be reused, but that might have been a bit YAGNI.  I think at this point it's as simple as it can get.  Hopefully we are good to merge now!